examples/perplexity/perplexity.cpp

```diff
@@ -406,13 +404,27 @@ static results_perplexity perplexity_v2(llama_context * ctx, const gpt_params &
             const int batch_start = start + j * n_batch;
             const int batch_size  = std::min(end - batch_start, n_batch);
 
+            llama_batch batch = llama_batch_init(batch_size, 0, 1);
+            for (int k = 0; k < batch_size; ++k) {
+                const int idx = batch_start + k;
+                batch.token  [k] = tokens[idx];
+                batch.output [k] = 1;
+            }
+            batch.n_tokens   = batch_size;
+            batch.pos        = nullptr;
+            batch.n_seq_id   = nullptr;
+            batch.seq_id     = nullptr;
+            batch.all_pos_0  = j*n_batch;
+            batch.all_pos_1  = 1;
+            batch.all_seq_id = 0;
```

Comment:
This could also be done simply with

```c
llama_batch_add(batch, tokens[idx], j*n_batch + k, { 0 }, true);
```

within the loop over `k`.

Reply:
Changes have been made.